hub/core/experimental/cleanlab.py

```diff
@@ -0,0 +1,253 @@
+            elif dataset[tensor].htype == "class_label":
+                labels_tensor = tensor
+    else:
+        for tensor in tensors_list:
+            if dataset[tensor].htype == "image":
+                images_tensor = tensor
+            elif dataset[tensor].htype == "class_label":
+                labels_tensor = tensor
+
+    if images_tensor and labels_tensor:
+        tensors = [images_tensor, labels_tensor]
+
+    else:
+        raise ValueError(
+            "Could not find the images and labels tensors. Please provide the images and labels tensors."
+        )
+
+    return tensors
+
+
+def estimate_cv_predicted_probabilities(
+    dataset, labels, model, folds, num_classes, verbose
+):
+    """
+    This function computes an out-of-sample predicted
+    probability for every example in a dataset using cross
+    validation. Output is an np.array of shape (N, K) where N is
+    the number of training examples and K is the number of classes.
+    """
+
+    # Initialize pred_probs array
+    pred_probs = np.zeros(shape=(len(dataset), num_classes))
+
+    # Create cross-validation object for out-of-sample predicted probabilities.
+    kfold = StratifiedKFold(n_splits=folds, shuffle=True)
+
+    if verbose:
+        print(
+            "Computing out-of-sample predicted probabilities with "
+            f"{folds}-fold cross validation..."
+        )
+
+    # Loop through each fold and compute out-of-sample predicted probabilities.
+    for fold, (train_idx, holdout_idx) in enumerate(kfold.split(X=dataset, y=labels)):
+
+        if verbose:
+            print(f"Fold {fold + 1} of {folds}...")
+
+        # Initialize a fresh untrained model.
+        model_copy = clone(model)
+
+        # Select the training and holdout cross-validated sets.
+        ds_train, ds_holdout = (
+            dataset[train_idx.tolist()],
+            dataset[holdout_idx.tolist()],
+        )
+        # Fit model to training set, predict on holdout set, and update pred_probs.
+        model_copy.fit(X=ds_train)
+
+        pred_probs_cross_val = model_copy.predict_proba(X=ds_holdout)
+        pred_probs[holdout_idx] = pred_probs_cross_val
+
+    return pred_probs
+
+def append_label_issues_tensors(dataset, label_issues, label_quality_scores):
+    """
+    This function creates a group of tensors label_issues.
+    After creating tensors, automatically commits the changes.
+    """
+    with dataset:
+
+        dataset.create_group("label_issues")
+        dataset.label_issues.create_tensor(
+            "is_label_issue", htype="generic", dtype="bool"
+        )
+        dataset.label_issues.create_tensor(
+            "label_quality_scores", htype="generic", dtype="float64"
+        )
+
+        for label_issue, label_quality_score in zip(label_issues, label_quality_scores):
+            dataset.label_issues.is_label_issue.append(label_issue)
+            dataset.label_issues.label_quality_scores.append(label_quality_score)
+
```

Comment:
I would probably need to add some checks here to ensure the tensors were written correctly. However, it seems that without the flag `skip_ok` you can't skip any tensors anyways, and if any append operation fails for an individual tensor, nothing will be appended at all.